core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -1463,4 +1472,59 @@ abstract class Bundle extends Record {
     *   the fields in the order they were defined
     */
   override def toPrintable: Printable = toPrintableHelper(_elements.toList.reverse)
+
+  private[chisel3] override def bind(target: Binding, parentDirection: SpecifiedDirection): Unit = {
+    super.bind(target, parentDirection)
+
+    this match {
+      case aliasedBundle: HasTypeAlias =>
+        aliasedBundle.aliasName.map(alias => {
+          // If the aliased bundle is coerced and it has flipped signals, then they must be stripped
+          val isFlipped = DataMirror
+            .collectMembers(this) { case d: Data if d.passiveDirection == SpecifiedDirection.Flip => d }
+            .toSeq
+            .nonEmpty
+          val isCoerced = direction match {
+            case ActualDirection.Input | ActualDirection.Output => true
+            case other                                          => false
```

Comment:
This logic to determine coercedness/strippedness should be integrated into the existing Record.bind code so that we don't have to iterate over the elements an additional time.

Furthermore, this needs to be done in a hierarchical way--DataMirror.collectMembers is recursive which means this code is effectively `O(n^d)` (where `n` is number of Bundle elements and `d` is the depth of your Bundle hierarchy), every single Bundle in a nested hierarchy is recursively accessing all fields so every field gets accessed many many times.

The purpose of adding a new field like `_passiveDirection` is so that the value is cached for each `Record` as you go up the tree and thus you don't need to access the children of any Records. Also, for Vecs you really just need to check the `sample_element`.

I'm not entirely convinced that `_passiveDirection` is the right information to be storing here so please think it through and figure out what information is needed so that this can be done live inside of `Record.bind` and not require touching children recursively.